### ISSUE TITLE
Could the nod explorer be exposed as a UI element?

### ISSUE THREAD

**Issue by XertroV** [openplanet-nl/openplanet-issues#304]
I'm wondering if something like `UI::ExploreNodInner(string, CMwNod@ nod)` is possible. Obvs this would only work in dev mode.

Basically I have some simple dev plugins to help with research, and one let's me look at the picked block. This is normally hard because it's only set when you hover, so I have to cache a reference to it (that's fine so far). It's also useful b/c `map.Blocks` is like always more than 2500 so finding a specific CGameCtnBlock to explore can be hard. 

If I explore this nod, tho, and quit the editor, then often it results in a crash since the nod gets cleared -- fair enough.

My dev plugin handles this case, though, and makes sure to clear references and things. If I could integrate the nod explorer for a specific nod into my own UI, then I can ensure it never renders when the editor is quit.

Additionally, this helps with deleting blocks, since I can manage ref counts in my own plugin to avoid the memory being cleared, but if I have it open in nod explorer at the same time then there's a risk of a crash.

**Comment by codecat** [comment]
Have you looked at [UI::NodTree](https://openplanet.dev/docs/api/UI/NodTree)?

**Comment by XertroV** [comment]
Yes, though I'd forgotten about it till now. The problem I ran into is that it's not possible to display values -- well unless I hardcode each type, I guess. 

Code like the following could be autogenerated based on `OpenplanetNext.json`, but it feels tedious and would be slower than a c++ impl. 

```angelscript
// use via `UI::NodTree(nod, NodTreeMemberCB);`

void NodTreeMemberCB(CMwNod@ nod, Reflection::MwMemberInfo@ info) {
    // callback for main nod
    if (info is null) return;
    bool handled = DrawNod_AnchoredObj(nod, info)
        || DrawNod_AnchoredObj(nod, info)
        // || DrawNod_X(nod, info)
        ;
    if (!handled) {
        UI::Text("Unhandled: ." + info.Name);
    }
}

bool DrawNod_AnchoredObj(CMwNod@ _item, Reflection::MwMemberInfo@ info) {
    auto item = cast<CGameCtnAnchoredObject>(_item);
    if (item is null) return false;
    if (info.Name == "BlockUnitCoord") {
        UI::Text(info.Name + " = " + item.BlockUnitCoord.ToString());
    } else if (info.Name == "AbsolutePositionInMap") {
        item.AbsolutePositionInMap = UI::InputFloat3("AbsolutePositionInMap", item.AbsolutePositionInMap);
    } else {
        UI::Text(info.Name);
    }
    return true;
}
```

![image](https://user-images.githubusercontent.com/1046448/235332376-1e521cb9-56fa-4260-abf5-df69f6dc2022.png)

**Comment by codecat** [comment]
I see. The Nod Explorer in itself is very large, but it uses the same NodTree code underneath. To make this work, I'd have to move all that code into some other generic place, and I'm not entirely sure if that's worth doing right now to be honest. We can keep this issue around though, it might make sense to do at some point.
